### PR TITLE
Crop operators cpu

### PR DESCRIPTION
**Pull request by drivanov**

### CODE REVIEW

File: dali/pipeline/operators/crop/crop.cc [hunk 1]
```diff
@@ -13,25 +13,167 @@
...
+}
+
+template<>
+template<typename Out>
+void Crop<CPUBackend>::RunHelper(SampleWorkspace *ws, const int idx) {
+  const auto &input = ws->Input<CPUBackend>(idx);
+  auto output = ws->Output<CPUBackend>(idx);
+
+  // Validate
+  if (output_type_ == DALI_FLOAT) {
```
Comment (JanuszL) [inline dali/pipeline/operators/crop/crop.cc]: This is done in RunImpl which calls RunHelper with proper type. So just
```ValidateHelper<Out>(&input, output);```
should do

File: dali/pipeline/operators/crop/crop.cu [hunk 2]
```diff
@@ -13,23 +13,201 @@
...
+  // Calculate input pointers and copy to gpu
+  for (int i = 0; i < batch_size_; ++i) {
+    input_ptrs_.template mutable_data<const uint8*>()[i] =
+      input.template tensor<uint8>(i) + crop_offsets_[i];
+  }
+  input_ptrs_gpu_.Copy(input_ptrs_, ws->stream());
+  input_strides_gpu_.Copy(input_strides_, ws->stream());
+
+  // Validate
+  if (output_type_ == DALI_FLOAT) {
```
Comment (JanuszL) [inline dali/pipeline/operators/crop/crop.cu]: Maybe we could just call ValidateHelper from RunHelper to avoid this if/else ladder?

**Review by JanuszL — COMMENTED** [review]
Rebase please as it cannot be merged now.

File: dali/pipeline/operators/crop/crop.cc (hunk 1 above)
Comment (drivanov) [reply dali/pipeline/operators/crop/crop.cc]: Yes, it's a good point!

**Comment by drivanov** [comment]
Re-created a new branch from scratch

**Pull request closed without merging**